#!/usr/bin/env python
PACKAGE = "echtzeitsysteme"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("K_P",     double_t,    0, "Gain P",  17.5,  0, 100)
gen.add("K_I",     double_t,    0, "Gain I",   3.5,  0,  25)
gen.add("K_D",     double_t,    0, "Gain D",  	.1,  0,   1)
gen.add("vel",     int_t,       0, "Velocity",   0,  -1000,   1000)
gen.add("steering",int_t,       0, "Steering",   0,  -1000,   1000)
gen.add("ctrlDist",int_t,       0, "ctrlDist",   0,  0,   150)

size_enum = gen.enum([ gen.const("Small",      int_t, 1, "A small constant"),
                       gen.const("Medium",     int_t, 10, "A medium constant"),
                       gen.const("Large",      int_t, 100, "A large constant"),
                       gen.const("ExtraLarge", int_t, 1000, "An extra large constant")],
                     "An enum to set size")

gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "echtzeitsysteme", "Controller"))
